schematicbrushreborn-core/src/main/java/de/eldoria/schematicbrush/config/sections/presets/DatabasePresetRegistry.java

```diff
@@ -0,0 +1,284 @@
+package de.eldoria.schematicbrush.config.sections.presets;
+
+import com.google.gson.Gson;
+import de.eldoria.eldoutilities.serialization.SerializationUtil;
+import de.eldoria.eldoutilities.simplecommands.TabCompleteUtil;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class DatabasePresetRegistry implements PresetRegistry {
+
+    private static final Gson GSON = new Gson();
+    private Map<UUID, PresetContainer> playerPresets = new HashMap<>();
+    private PresetContainer globalPresets = new PresetContainerImpl();
+    private final DataSource dataSource;
+    private final Plugin plugin;
+
+    public DatabasePresetRegistry(DataSource dataSource, Plugin plugin) {
+        this.dataSource = dataSource;
+        this.plugin = plugin;
+        this.reload();
+    }
+
+    private void reload() {
+        this.playerPresets.clear();
+        this.globalPresets.getPresets().clear();
+        CompletableFuture.runAsync(() -> {
+            try(Connection connection = this.dataSource.getConnection()) {
+                try(PreparedStatement preparedStatement = connection.prepareStatement("CREATE TABLE IF NOT EXISTS player_presets (PLAYER_UUID longtext, PLAYER_PRESET longtext)")) {
+                    preparedStatement.execute();
+                }
+
+                try(PreparedStatement preparedStatement = connection.prepareStatement("CREATE TABLE IF NOT EXISTS global_presets (GLOBAL_PRESET longtext)")) {
+                    preparedStatement.execute();
+                }
+
+                try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT PLAYER_UUID, PLAYER_PRESET FROM player_presets")) {
+                    ResultSet resultSet = preparedStatement.executeQuery();
+
+                    while(resultSet.next()) {
+                        UUID uuid = UUID.fromString(resultSet.getString("PLAYER_UUID"));
+                        Preset preset = this.fromJson(resultSet.getString("PLAYER_PRESET"));
```

Comment:
While this might work now, GSON has a lot of trouble serializing and de serializing bukkit object.
Its mostly better to use the serialize() method on the object and dump the returned map into a json string. Same the other way around. Convert the json to a object map and pass the map into the object constructor.
Thats saver and will work with bukkit objects all the time.